tt_eager/tt_dnn/op_library/unpad/unpad_op.cpp

```diff
@@ -19,17 +19,80 @@ namespace tt {
 namespace tt_metal {
 
 
+uint32_t get_upper_dims_compressed(const Tensor & tensor) {
+
+    uint32_t upper_dims_compressed = 1;
+    for(int dim=0; dim<tensor.shape().rank() - 2; dim++) {
+        upper_dims_compressed *= tensor.shape()[dim];
+    }
+    return upper_dims_compressed;
+}
+
+uint32_t get_upper_start_offset(const Tensor & tensor, const Shape & output_tensor_start) {
+
+    uint32_t num_pages = tensor.volume();
+    if(tensor.layout() == Layout::TILE) {
+        num_pages /= (TILE_HW);
+    }
+    else {
+        uint32_t page_width = tensor.shape()[-1];
+        num_pages /= page_width;
+    }
+    // offset for every dim except last 2
+    uint32_t start_offset = 0;
+    for(int dim_outer=0; dim_outer<tensor.shape().rank() - 2; dim_outer++) {
+        uint32_t compressed_dims = 1;
+        for(int dim_inner = dim_outer; dim_inner >= 0; dim_inner--) {
```

Comment:
Can we use unsigned ints for consistency in the loop vars since rank() returns an unsigned type?

Reply:
that makes sense, just did that.